### PR TITLE
Added important containers and algorithms in cpp.md

### PR DESCRIPTION
**Pull request by sumitkr2000**
Hello @onecompiler-ops, this is pull request with the reference to issue no. #66.

I hade made some changes in cpp.md file. I have added important containers and algorithms which is present in C++ standard template library. This will surely benefit the visitors to your website and add some more knowledge to them. Hope you find this useful. 

Please review and merge my pull request. Thank you!

### CODE REVIEW

File: cpp.md [hunk 1]
```diff
@@ -9,9 +9,10 @@ C++ is a widely used middle-level programming language which is used in developi
 
 ## Basics
 * `cin >> x`: read value into the variable x from input stream
-* `cout << x`: printf value to the output stream
+* `cout << x`: print value to the output stream
 * `//`: single line comments
 * `/* */`: Multi line comments
+* `endl`: new line
```
Comment (onecompiler-ops) [inline cpp.md:15]: This is not widely used, please remove

File: cpp.md [hunk 2]
```diff
@@ -27,6 +28,7 @@ int main() {
 * `cout`: to print the output.
 * `main()`: main function is the entry point of any C++ program.
 * `return 0`: To end the main function
+* `endl`: to change line of the output
```
Comment (onecompiler-ops) [inline cpp.md:31]: This is not widely used, please remove

**Review by onecompiler-ops — CHANGES_REQUESTED** [review]

File: cpp.md (hunk 2 above)
Comment (sumitkr2000) [reply cpp.md]: @onecompiler-ops  I have removed "endl" from the 2 places where I had added it. Please review it and merge my pull request. Thank you!

File: cpp.md [hunk 3]
```diff
@@ -379,43 +434,203 @@ int main() {
...
+    }
+
+    for (auto i: mp) {
+        cout << " " << i.first << " " << i.second << endl;
+    }
+}
+
+```
+
+## Algorithms
```
Comment (onecompiler-ops) [inline cpp.md:574]: Can you make this as a Markdown table to represent in a concise way?

File: cpp.md [hunk 4]
```diff
@@ -274,8 +278,56 @@ typedef data-type name;
...
+vector<int> v;
+
+v = {3,2,1};
+
+sort(v.begin(), v.end()); // sorting in the vector
+
+}
+
+```
+## Header Files
```
Comment (onecompiler-ops) [inline cpp.md:308]: These changes are there, can you rebase?

File: cpp.md [hunk 5]
```diff
@@ -274,8 +278,56 @@ typedef data-type name;
 typedef unsigned int distance;  // typedef of int 
 ```
 
-## Vectors
-Vectors are same as dynamic arrays. They will be resized on element inseertions & deletions.
+## Sort one-line
```
Comment (onecompiler-ops) [inline cpp.md:281]: We want to cover basics only in Cheat Sheets, this is advanced information. You can always create these as posts in OneCompiler

**Review by onecompiler-ops — CHANGES_REQUESTED** [review]

File: cpp.md (hunk 3 above)
Comment (sumitkr2000) [reply cpp.md:574]: @onecompiler-ops sure why not! I guess I had made it lengthy !

File: cpp.md (hunk 5 above)
Comment (sumitkr2000) [reply cpp.md:281]: @onecompiler-ops which things should i remove from cheatsheet please elaborate...I will make it better. Thank you!

File: cpp.md (hunk 4 above)
Comment (sumitkr2000) [reply cpp.md:308]: @onecompiler-ops rebase means?

**Comment by sumitkr2000** [comment]
Hey @onecompiler-ops Idk how this pull request automatically got closed . Can you please open it?

**Comment by onecompiler-ops** [comment]
Screenshot 2022-10-03 at 1 05 27 PM

**Comment by sumitkr2000** [comment]
@onecompiler-ops can u suggest me the changes i have to make in my commits. I will make another pull request and submit my changes.

**Pull request closed without merging**